Base/Python/slicer/util.py

```diff
@@ -2322,7 +2325,10 @@ def warningDisplay(text, windowTitle=None, parent=None, standardButtons=None, **
   """Display popup with a warning message.
 
   If there is no main window then the text is only logged (at warning level).
+  Skipped during testing.
   """
+  if slicer.app.testingEnabled():
+    return
```

Comment:
To make sure the test effectively covers code that would otherwise be executed, I suggest to move

```python
  if slicer.app.testingEnabled():
    logging.info("Testing mode is enabled: Skipping warning popup [%s]" % windowTitle)
    return
```

just above the call to `messageBox` below.

Doing so would also make sure `logging.warning(text)` is called.